cmd/doracled/cmd/gen_oracle_key.go

```diff
@@ -61,14 +55,15 @@ So please be cautious in using this command.`,
 		}
 
 		// seal and store oracle private key
-		if err := sgx.SealToFile(oraclePrivKey.Serialize(), oraclePrivKeyPath, conf.Enclave.Enable); err != nil {
+		if err := sgx.SealToFile(oraclePrivKey.Serialize(), oraclePrivKeyPath); err != nil {
 			log.Errorf("failed to write %s: %v", oraclePrivKeyPath, err)
 			return err
 		}
 
 		// generate oracle key remote report
 		oraclePubKey := oraclePrivKey.PubKey().SerializeCompressed()
-		oracleKeyRemoteReport, err := sgx.GenerateRemoteReport(oraclePubKey[:], conf.Enclave.Enable)
+		oraclePubKeyHash := sha256.Sum256(oraclePubKey)
```

Comment:
Is there a reason you hashed this?
Is it because of the data length limit of the report?

Reply:
Yes. However, since the length of pubkey is also fixed, we can use the pubkey as is.

Reply:
To make the `sgx.GenerateRemoteReport` more general, it can be improved as below,

```go
func GenerateRemoteReport(data []byte) ([]byte, error) {
        hashedData := sha256.Sum256(data)
	return enclave.GetRemoteReport(hashedData)
}
```

No matter how long the length of data is, this function can generate a valid remote report.

Reply:
IIRC, the remote report can contain max 512 bits. So, you can use sha512.

I would prefer executing hash functions outside of the `GenerateRemoteReport` explicitly.

Reply:
Agreed. And I'm thinking not using hash of public key. There is no need to hash the public key because its length is shorter than _maxReportData_(64).